homeassistant/components/sensor/mqtt.py

```diff
@@ -73,6 +81,9 @@ def async_added_to_hass(self):
         """
         @callback
         def message_received(topic, payload, qos):
+            """Reset expiration time."""
+            self._value_expiration_at = time.time() + self._expire_after
```

Comment:
Please use the event helper `async_track_point_in_utc_time` to call an async function to set the state to None. `async_track_point_in_utc_time` will return a function that can be called to remove the listener. So on message received, remove the last listener and add a new time listener.